mesonbuild/build.py

```diff
@@ -402,6 +402,11 @@ def process_kwargs(self, kwargs):
             self.build_by_default = kwargs['build_by_default']
             if not isinstance(self.build_by_default, bool):
                 raise InvalidArguments('build_by_default must be a boolean value.')
+        elif 'install' in kwargs and isinstance(kwargs['install'], bool) and kwargs['install']:
```

Comment:
`install` is always supposed to be a bool, if it isn't then we have a bug. Therefore, I think you can simplify this to:
```python
elif kwargs.get('install', False):
     self.build_by_default = True
```
Since we know that install is a boolean, and that it is `True`, just set build_by_default to `True`.

Reply:
makes sense, done